### PR TITLE
Implemented flyout closure on navigation.

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
Addresses blackbaud/skyux-flyout#20

### CODE REVIEW

File: src/app/public/modules/flyout/flyout.service.ts [hunk 1]
```diff
@@ -5,6 +5,10 @@ import {
   OnDestroy
 } from '@angular/core';
 
+import {
+  Router, NavigationStart
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/flyout/flyout.service.ts]: imports should be on a new line

Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/flyout/flyout.service.ts]: Grrrr auto import. Fixed

File: src/app/public/modules/flyout/flyout.service.ts [hunk 2]
```diff
@@ -45,7 +50,8 @@ export class SkyFlyoutService implements OnDestroy {
   constructor(
     private adapter: SkyFlyoutAdapterService,
     private windowRef: SkyWindowRefService,
-    private dynamicComponentService: SkyDynamicComponentService
+    private dynamicComponentService: SkyDynamicComponentService,
+    private router: Router
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/flyout/flyout.service.ts]: Do we also need to import the `RouterModule` into `SkyFlyoutModule`?

File: src/app/public/modules/flyout/flyout.service.spec.ts [hunk 3]
```diff
@@ -39,6 +45,7 @@ import {
 describe('Flyout service', () => {
   let service: SkyFlyoutService;
   let applicationRef: ApplicationRef;
+  let router: Router;
 
   beforeEach(() => {
     TestBed.configureTestingModule({
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/flyout/flyout.service.spec.ts]: I seem to remember that it's good practice to import the `RouterTestingModule` into the `TestBed`. 
https://angular.io/api/router/testing/RouterTestingModule

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/flyout/flyout.service.spec.ts]: Might need it in the flyout component spec, too.

File: src/app/public/modules/flyout/flyout.service.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/flyout/flyout.service.ts]: It was already there :)

File: src/app/public/modules/flyout/flyout.service.spec.ts (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/flyout/flyout.service.spec.ts]: It is :) It is imported via the `SkyFlyoutFixturesModule` :)

**Comment by Blackbaud-TrevorBurch** [comment]
Good call on the screenshot names @Blackbaud-SteveBrush that is done :)

**Pull request merged**